### PR TITLE
Feature Request 17: Implement contextual data

### PR DESCRIPTION
**Pull request by crazycodr**
Refer to #17

### CODE REVIEW

File: Helpers/WithContext.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+namespace Exceptions\Helpers;
+
+/**
+ * Implements a set of helper methods that creates exceptions to attach and retrieve contextual data.
+ *
+ * @package Exceptions\Helpers
+ */
+trait WithContext
+{
+    protected $context = null;
```
Comment (Gounlaf) [inline Helpers/WithContext.php:17]: Missing phpdoc?

```php
/** @var mixed */
protected $context = null;
```

Comment (crazycodr) [reply Helpers/WithContext.php:17]: If thats all, i'll merge right after

Comment (Gounlaf) [reply Helpers/WithContext.php:17]: That's all I see; other changes you made seems to be OK for me :)

Comment (Gounlaf) [reply Helpers/WithContext.php:17]: Well... I just saw something might be missing: the previous exception =/

**Pull request merged**